Gems/GradientSignal/Code/Tests/GradientSignalBenchmarks.cpp

```diff
@@ -160,6 +168,195 @@ namespace UnitTest
     GRADIENT_SIGNAL_GET_VALUES_BENCHMARK_REGISTER_F(GradientGetValues, BM_SurfaceMaskGradient);
     GRADIENT_SIGNAL_GET_VALUES_BENCHMARK_REGISTER_F(GradientGetValues, BM_SurfaceSlopeGradient);
 
+    // --------------------------------------------------------------------------------------
+    // Gradient Surface Data
+
+
+    class GradientSurfaceData : public GradientSignalBenchmarkFixture
+    {
+    public:
+        /* To benchmark the GradientSurfaceDataComponent, we need to create a surface provider in the world, then use
+           the GradientSurfaceDataComponent to modify the surface points.
+
+           For the surface provider, we create a flat box centered in XY that's the XY size of the world.
+           For the GradientSurfaceDataComponent, we'll use a constant gradient as its input, and a sphere centered in XY that's
+           the XY size of the world as its constrained bounds.
+
+           Every surface point within the sphere will have the tags from the provider and the modifier, and every point outside the
+           sphere will only have the provider tags.
+        */
+        AZStd::vector<AZStd::unique_ptr<AZ::Entity>> CreateBenchmarkEntities(float worldSize)
+        {
+            AZStd::vector<AZStd::unique_ptr<AZ::Entity>> testEntities;
+            float halfWorldSize = worldSize / 2.0f;
+
+            // Create a large flat box with 2 provider tags.
+            {
+                AZStd::unique_ptr<AZ::Entity> surface = AZStd::make_unique<AZ::Entity>();
+                AZ::Vector3 worldPos(halfWorldSize, halfWorldSize, 10.0f);
+
+                auto transform = surface->CreateComponent<AzFramework::TransformComponent>();
+                transform->SetLocalTM(AZ::Transform::CreateTranslation(worldPos));
+                transform->SetWorldTM(AZ::Transform::CreateTranslation(worldPos));
```

Comment:
nit: I'm pretty sure when you set the world tm, it also sets the local, so the above call is probably unnecessary.

Reply:
Removed all extra SetLocalTM calls.